src/Flux/utils.jl

```diff
@@ -1,16 +1,25 @@
 function destructure(m)
```

Comment:
@MikeInnes can we make this a utility in Flux instead of DiffEqFlux? 
@ChrisRackauckas idk if you remember but on my fork of DiffEqFlux I have destructure/restructure returning slightly different objects and I think we might prefer them. Essentially have `destructure` return the vector and an anonymous function which `restructure`s. Removes the need for a separate `restructure` function and also that dependence on `m`, which is a little confusing, since it only uses `m` for its structure, not its current parameter values. What do you think?

Reply:
I had a similar plan of action in mind, with the params wrapped in a `Buffer`. The anonymous function sounds about right too. Can we `loadparams!` on `m` by any chance? That would be cleaner, but might need some thought.